mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1433,7 +1442,7 @@ private static int doReplaceOne(final MongoDBQueryDescriptor queryDescriptor, fi
 			collation = ( col != null ) ? getCollation( col ) : null;
 		}
 
-		final UpdateOptions updateOptions = new UpdateOptions().upsert( upsert ).collation( collation );
+		final ReplaceOptions updateOptions = new ReplaceOptions().upsert( upsert ).collation( collation );
```

Comment:
maybe we can change variable name in `replaceOptions` to be consistent with its type